Moq.AutoMock/Resolvers/DelegateResolver.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Reflection;
+
+namespace Moq.AutoMock.Resolvers
+{
+    public class DelegateResolver : IMockResolver
+    {
+        public void Resolve(MockResolutionContext context)
+        {
+            if (context == null) return;
+
+            var (am, serviceType, value) = context;
+            if (!typeof(Delegate).GetTypeInfo().IsAssignableFrom(serviceType.GetTypeInfo()))
```

Comment:
This would resolve against any delegate type. Though this may seem appealing I will argue this is not what we want. I think we want to be much more restrictive and only resolve Func.

Reply:
Hmmm, as far as mocking the return value goes, it seems like `Func` should just be a special case of a `Delegate`. I would be more inclined to agree if this were included by default (as opposed to an opt-in) Should we add `FuncResolver` by default and leave this one for a user to include?

Reply:
I don't think it matters if this is an opt-in behavior or not. I don't think matching on all delegate types is the correct behavior.

Consider this the case where I have written something like
`public delegate T Coerce<T>()`
I would not want AM to attempt to resolve it. Despite having an identical signature to `Func<T>` I would not expect AM To resolve it unless I have explicitly included it with `.Use` or written my own behavior.